test/spec/test_spec.rb

```diff
@@ -0,0 +1,44 @@
+require_relative "spec_helper"
+
+describe "A Java application" do
+
+  it "works with the getting started guide" do
+    Hatchet::Runner.new("java-getting-started", buildpacks: ["https://github.com/heroku/heroku-buildpack-java"]).tap do |app|
+      app.deploy do
+        #deploy works
+      end
+    end
+  end
+
+  it "checks for bad version" do
+    Hatchet::Runner.new("java-getting-started", buildpacks: ["https://github.com/heroku/heroku-buildpack-java"], allow_failure: true).tap do |app|
+      app.before_deploy do
+        set_java_version(Dir.pwd, "14.badversion")
+      end
+      app.deploy do
+        expect(app.output).to include("ERROR: Unsupported Java version: 14.badversion")
+      end
+    end
+  end
+
+  it "has absolute buildpack paths" do
+    buildpacks = [
+      :default,
```

Comment:
I didn't catch this before. Why do we need `:default` here? The other tests use the buildpack URL directly.

Reply:
:default is used wherever you want the java buildpack inserted, it will be replaced with the repo and git branch that is being tested.

Reply:
Lillian is correct, it will replace `:default` with `https://github.com/heroku/heroku-buildpack-jvm-common#<branchname>`. The interface is not great. I'm open to suggestions for alternate interfaces.

Reply:
Shouldn't the other tests then use the same? If I understand correctly, the other tests will then use `main` which defeats the purpose of these tests. Is that correct?

Reply:
By default all tests use `https://github.com/heroku/heroku-buildpack-jvm-common#<branchname>` but when you're specifying the `buildpacks:` key we need some way to specify where in the array you want your buildpack to execute.

Previously you had to do something like 

```
[
    HATCHET_BUILDPACK_BASE + Hatchet.git_branch,
      "https://github.com/sharpstone/force_absolute_paths_buildpack"
    ]
```

Which I could never remember and had to keep looking it up, so I added this `:default` shorthand.

Reply:
Okay. I see now. These tests need to be updated so they're using the branch being modified 

```
Hatchet::Runner.new("java-getting-started", buildpacks: ["https://github.com/heroku/heroku-buildpack-java"]).run_ci do |test_run| 
```

Needs to be 

```
     Hatchet::Runner.new("java-getting-started").run_ci do |test_run| 
```

Reply:
Wouldn't doing this cause the buildpack to default to the ruby buildpack?

Reply:
Oh nevermind, the java buildpack is set in the circleci environment

Reply:
@lillianzhang331 Can we get the change ready so we can merge this in?